lib/drivers/ism43340/ism43340.cpp

```diff
@@ -0,0 +1,270 @@
+#include "Assert.hpp"
+#include "Logger.hpp"
+
+#include <memory>
+
+#define CMD_SET_HUMAN_READABLE "$$$"
+#define CMD_SET_MACHINE_READABLE "---"
+#define CMD_RESET_SETTINGS "Z0"
+
+#define CMD_JOIN_NETWORK "C0"
+#define CMD_SET_SSID "C1="
+#define CMD_SET_PASSWORD "C2="
+#define CMD_SET_SECURITY "C3="
+#define CMD_SET_DHCP "C4="
+#define CMD_GET_CONNECTION_INFO "C?"
+
+#define CMD_SET_TRANSPORT_PROTOCOL "P1="
+#define CMD_SET_HOST_IP "P3="
+#define CMD_SET_PORT "P4="
+#define CMD_START_CLIENT "P6="
+
+#define CMD_SEND_DATA "S3="
+
+Ism43340::Ism43340(SpiPtrT sharedSPI, PinName nCs, PinName intPin,
+                   PinName _nReset)
+    : CommLink(sharedSPI, nCs, intPin), nReset(_nReset) {
+    reset();
+}
+
+inline void Ism43340::formatCommand(std::string* command) {
+    if (command->size() % 2 == 0) {
+        command->append("\r\n");
+    } else {
+        command->append("\r");
+    }
+}
+
+int Ism43340::writetospi(std::string* command) {
+    while (dataready.read() != 1) {
+    }
+
+    uint8_t lastc = 0;
+    radiospi.chipselect();
+    for (int i = 0; i < command->size(); i++) {
+        uint8_t c = command[i];
+        if (lastc != 0) {
+            uint16_t packet = (c << 8) | lastc;
+            radiospi.m_spi->write(packet);
```

Comment:
Is there a multi write byte? There may be some underlying differences in performance

Reply:
I think this is an endianness isuees. The radio is the opposite endianness of the of mbed. I believe there is a write_8 and write_16 (hidden) but the endianness is wrong and the hardware module doesn't support switching.

Reply:
Now that my life is starting to actually come back together somewhat I just wanted to comment on some of these. Multibyte write through the mbeds spi api only supports 8 bit transfers and as such wont do any endianness conversion.